### PR TITLE
[Java] Added splicing example for IntVector

### PR DESCRIPTION
**Pull request by mystic-lama**
Added an example to show splicing on IntVector

### CODE REVIEW

File: java/source/create.rst [hunk 1]
```diff
@@ -187,4 +187,40 @@ Array of List
...
+                vector.setSafe(i, i);
+            }
+            vector.setValueCount(10);
+
+            TransferPair tp = vector.getTransferPair(allocator);
+            tp.splitAndTransfer(0, 5);
+            IntVector sliced = (IntVector) tp.getTo();
+            System.out.print(sliced);
+            sliced.clear();
+        }
```
Comment (davisusanibar) [inline java/source/create.rst:222]:
```suggestion
    try (BufferAllocator allocator = new RootAllocator();
         IntVector vector = new IntVector("intVector", allocator)) {
        for (int i = 0; i < 10; i++) {
            vector.setSafe(i, i);
        }
        vector.setValueCount(10);

        TransferPair tp = vector.getTransferPair(allocator);
        tp.splitAndTransfer(0, 5);
        try (IntVector sliced = (IntVector) tp.getTo()) {
            System.out.print(sliced);
        }
    }
```

**Comment by davisusanibar** [comment]
Thank you @mystic-lama for your support

File: java/source/create.rst [hunk 2]
```diff
@@ -187,4 +187,40 @@ Array of List
 
 .. _`FieldVector`: https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/FieldVector.html
 .. _`ValueVector`: https://arrow.apache.org/docs/java/vector.html
-.. _`dictionary-encoding`: https://arrow.apache.org/docs/format/Columnar.html#dictionary-encoded-layout
\ No newline at end of file
+.. _`dictionary-encoding`: https://arrow.apache.org/docs/format/Columnar.html#dictionary-encoded-layout
```
Comment (davisusanibar) [inline java/source/create.rst]: Only by readability could be possible to move these lines to end of file

File: java/source/create.rst [hunk 3]
```diff
@@ -187,4 +187,40 @@ Array of List
...
+
+Splicing
+========
+
+Splicing provides a way of copying complete or partial data between same types of Vectors
```
Comment (lidavidm) [inline java/source/create.rst:195]:
```suggestion
Splicing provides a way of copying a range of rows between two vectors of the same type.
```

File: java/source/create.rst [hunk 4]
```diff
@@ -187,4 +187,40 @@ Array of List
...
+
+Splicing IntVector
+------------------
+
+In this example, we want to copy a portion of input Vector to a new Vector.
```
Comment (lidavidm) [inline java/source/create.rst:200]:
```suggestion
In this example, we copy a portion of the input IntVector to a new IntVector.
```

File: java/source/create.rst (hunk 1 above)
Comment (lidavidm) [reply java/source/create.rst:222]: Maybe show slicing from a non-zero index?

Comment (mystic-lama) [reply java/source/create.rst:222]: @lidavidm more examples coming

File: java/source/create.rst (hunk 2 above)
Comment (mystic-lama) [reply java/source/create.rst]: Sure thing, let me move them to end and update the PR. Thanks

Comment (mystic-lama) [reply java/source/create.rst]: updated

File: java/source/create.rst [hunk 5]
```diff
@@ -185,6 +185,43 @@ Array of List
...
+Splicing provides a way of copying a range of rows between two vectors of the same type.
+
+Splicing IntVector
+------------------
+
+In this example, we copy a portion of the input IntVector to a new IntVector.
+
+.. testcode::
+
+    import org.apache.arrow.memory.BufferAllocator;
```
Comment (lidavidm) [inline java/source/create.rst:201]: Just one thing: everything here is indented by one space too many I think (it should line up). Same with the output below

Comment (mystic-lama) [reply java/source/create.rst:201]: There is similar comment on other PR, shall fix this once the other is fixed as expected :)

Comment (lidavidm) [reply java/source/create.rst:201]: The other PR is now merged

Comment (mystic-lama) [reply java/source/create.rst]: Done, hopefully it shall be correct this time.

**Review by lidavidm — APPROVED** [review]

**Pull request merged**